src/sql/workbench/services/connection/browser/connectionDialogService.ts

```diff
@@ -235,14 +235,15 @@ export class ConnectionDialogService implements IConnectionDialogService {
 			return Promise.resolve();
 		}
 		let fromEditor = params && params.connectionType === ConnectionType.editor;
+		let fromExtension = params && params.connectionType === ConnectionType.extension;
 		let uri: string = undefined;
 		if (fromEditor && params && params.input) {
 			uri = params.input.uri;
 		}
 		let options: IConnectionCompletionOptions = this._options || {
 			params: params,
-			saveTheConnection: true,
-			showDashboard: params && params.showDashboard !== undefined ? params.showDashboard : !fromEditor,
+			saveTheConnection: !fromExtension,
```

Comment:
I'm not sure why "fromExtension" means don't save?

Reply:
changed the ConnectionType name to `temporary` to reflect the same.